src/character.cpp

```diff
@@ -5381,77 +5383,76 @@ void Character::update_bodytemp()
+        temp_conv[bp->token] += sunlight_warmth;
         // DISEASES
-        if( bp == bp_head && has_effect( effect_flu ) ) {
-            temp_conv[bp] += 1500;
+        if( bp == bodypart_id( "head" ) && has_effect( effect_flu ) ) {
+            temp_conv[bp->token] += 1500;
         }
         if( has_common_cold ) {
-            temp_conv[bp] -= 750;
+            temp_conv[bp->token] -= 750;
         }
         // Loss of blood results in loss of body heat, 1% bodyheat lost per 2% hp lost
-        temp_conv[bp] -= blood_loss( bp ) * temp_conv[bp] / 200;
+        temp_conv[bp->token] -= blood_loss( bp->token ) * temp_conv[bp->token] / 200;
 
         // EQUALIZATION
-        switch( bp ) {
-            case bp_torso:
-                temp_equalizer( bp_torso, bp_arm_l );
-                temp_equalizer( bp_torso, bp_arm_r );
-                temp_equalizer( bp_torso, bp_leg_l );
-                temp_equalizer( bp_torso, bp_leg_r );
-                temp_equalizer( bp_torso, bp_head );
-                break;
-            case bp_head:
-                temp_equalizer( bp_head, bp_torso );
-                temp_equalizer( bp_head, bp_mouth );
-                break;
-            case bp_arm_l:
-                temp_equalizer( bp_arm_l, bp_torso );
-                temp_equalizer( bp_arm_l, bp_hand_l );
-                break;
-            case bp_arm_r:
-                temp_equalizer( bp_arm_r, bp_torso );
-                temp_equalizer( bp_arm_r, bp_hand_r );
-                break;
-            case bp_leg_l:
-                temp_equalizer( bp_leg_l, bp_torso );
-                temp_equalizer( bp_leg_l, bp_foot_l );
-                break;
-            case bp_leg_r:
-                temp_equalizer( bp_leg_r, bp_torso );
-                temp_equalizer( bp_leg_r, bp_foot_r );
-                break;
-            case bp_mouth:
-                temp_equalizer( bp_mouth, bp_head );
-                break;
-            case bp_hand_l:
-                temp_equalizer( bp_hand_l, bp_arm_l );
-                break;
-            case bp_hand_r:
-                temp_equalizer( bp_hand_r, bp_arm_r );
-                break;
-            case bp_foot_l:
-                temp_equalizer( bp_foot_l, bp_leg_l );
-                break;
-            case bp_foot_r:
-                temp_equalizer( bp_foot_r, bp_leg_r );
-                break;
-            default:
-                debugmsg( "Wacky body part temperature equalization!" );
-                break;
+        if( bp == bodypart_id( "torso" ) ) {
+            temp_equalizer( bodypart_id( "torso" ), bodypart_id( "arm_l" ) );
+            temp_equalizer( bodypart_id( "torso" ), bodypart_id( "arm_r" ) );
+            temp_equalizer( bodypart_id( "torso" ), bodypart_id( "leg_l" ) );
+            temp_equalizer( bodypart_id( "torso" ), bodypart_id( "leg_r" ) );
+            temp_equalizer( bodypart_id( "torso" ), bodypart_id( "head" ) );
+        }
+        if( bp == bodypart_id( "head" ) ) {
```

Comment:
it's a little ugly, but this should probably be a series of `else if` statements for a mild improvement in performance.

I'm not sure if it's worth predeclaring the bodypart_ids as consts.

Ideally, all this would go away and be replaced by lists of bodypart_ids to be equalized attached to each bodypart_id, but that's obviously beyond the scope of this PR.

Reply:
An alternate intermediate solution would be to have e.g. a hardcoded `unordered_map<bodypart_id, vector<bodypart_id>>`.  That might lead to cleaner code, but I don't think it matters that much.